python/doc/examples/meta_modeling/polynomial_chaos_metamodel/plot_chaos_conditional_expectation.py

```diff
@@ -0,0 +1,668 @@
+# the set of corresponding (un-reduced) multi-indices of
+# dimension :math:`\physicalInputDimension`:
+#
+# .. math::
+#    :label: PCE_CE_3
+#
+#     \cJ_{\overline{\vect{\alpha}}}
+#     = \left\{\alpha \in \cJ^P \; |\; \overline{\vect{\alpha}}_i
+#     = \alpha_{\overline{\vect{u}}_i}, \; i = 1, ..., |\overline{\vect{u}}|\right\}.
+#
+# Each aggregated coefficient :math:`\overline{a}_{\overline{\vect{\alpha}}} \in \Rset`
+# is defined by the equation:
+#
+# .. math::
+#    :label: PCE_CE_5
+#
+#     \overline{a}_{\overline{\vect{\alpha}}}
+#     = \sum_{\boldsymbol{\alpha} \in \cJ^P_{\overline{\vect{\alpha}}}}
+#     a_{\boldsymbol{\alpha}} \left(\prod_{i = 1}^{|\vect{u}|}
+#     \pi_{\alpha_{u_i}}^{(u_i)}\left(\standardInputObservation_{u_i}^{(0)}\right) \right).
+#
+# Finally:
+#
+# .. math::
+#    :label: PCE_CE_4
+#
+#     \widehat{h}(\standardInputObservation_{\overline{\vect{u}}})
+#     = \sum_{\overline{\boldsymbol{\alpha}} \in \overline{\cJ}}
+#     \overline{a}_{\overline{\vect{\alpha}}}
+#     \overline{\psi}(z_{\overline{\vect{u}}})
+#
+# for any :math:`\standardInputObservation_{\overline{\vect{u}}} \in \Rset^{|\overline{\vect{u}}|}`.
+
+# %%
+# The method is the following.
+#
+# - Create the reduced polynomial basis from equation :eq:`PCE_CE_1`.
+# - Create the list of reduced multi-indices from the equation :eq:`PCE_CE_2`, and, for each
+#   reduced multi-index, the list of corresponding multi-indices from the equation :eq:`PCE_CE_3`.
+# - Aggregate the coefficients from the equation :eq:`PCE_CE_5`.
+# - The parametric PCE is defined by the equation :eq:`PCE_CE_4`.
+
+# %%
+# Conditional expectation
+# ~~~~~~~~~~~~~~~~~~~~~~~
+#
+# One method to reduce the input dimension of a function is to
+# consider its conditional expectation.
+# The conditional expectation function is:
+#
+# .. math::
+#
+#   \widehat{\model}(\physicalInputObservation_{\vect{u}})
+#   = \Expect{\model(\physicalInputObservation)
+#            \; | \; \physicalInputRandomVector_{\vect{u}}
+#            = \physicalInputObservation_{\vect{u}}}
+#
+# for any :math:`\physicalInputObservation_{\vect{u}} \in \Rset^{|\vect{u}|}`.
+# In general, there is no dedicated method to create such a conditional expectation
+# in the library.
+# We can, however, compute efficiently the conditional expectation of a polynomial
+# chaos expansion.
+# In turn, this conditional chaos expansion (PCE) is a polynomial chaos expansion
+# which can be computed using the :meth:`~openturns.FunctionalChaosResult.getConditionalExpectation`
+# method from the :class:`~openturns.FunctionalChaosResult` class.
+
+# %%
+# Create the PCE
+# ~~~~~~~~~~~~~~
+
+# %%
+import openturns as ot
+import openturns.viewer as otv
+from openturns.usecases import ishigami_function
+import matplotlib.pyplot as plt
+
+# %%
+# The next function creates a parametric PCE based on a
+# given PCE and a set of indices.
+
+
+# %%
+def MeanParametricPCE(chaosResult, indices):
```

Comment:
This name is wrongly chosen, because a function name should have a lower case first letter.

But, there is a particular situation with OpenTURNS which is a C++ library. There are two naming which are mentioned in the [PEP8 naming section](https://peps.python.org/pep-0008/#descriptive-naming-styles) :
- the CamelCase,
- the snake_case.

According to the PEP8, function names should use the snake case ([see here](https://peps.python.org/pep-0008/#function-and-variable-names)), which is consistent to your suggestion. But OpenTURNS is a C++ library, where the method names have the CamelCase. Hence, in all my script, I now stick to the CamelCase. This way any Python code is consistent to the equivalent C++ counterpart.

I admit, however, that there is no consistency in the naming in the examples section. Searching for "def ", I found:
- [this example](http://openturns.github.io/openturns/latest/auto_reliability_sensitivity/reliability/plot_axial_stressed_beam.html#conclusion) uses the CamelCase for the functions `computeLogRelativeError` and `printMethodSummary`.
- [this other example](http://openturns.github.io/openturns/latest/auto_data_analysis/sample_analysis/plot_src_confidence.html) uses the snake case for the function name `draw_importance_factors_with_bounds`.

I guess that we may rename all functions in the doc, or none. I create the issue #2453 on this topic so that we can discuss this.

Reply:
Thanks for your research, you can resolve this discussion.